src/bootstrap/dist.rs

```diff
@@ -98,6 +98,137 @@ pub fn docs(build: &Build, stage: u32, host: &str) {
     }
 }
 
+fn find_files(files: Vec<&str>, path: &Vec<PathBuf>) -> Vec<PathBuf> {
+    let mut found = Vec::new();
+
+    for file in &files {
+        let file_path =
+            path .iter()
+                .map(|dir| dir.join(file))
+                .find(|p| fs::metadata(&p).is_ok());
+
+        if let Some(file_path) = file_path {
+            found.push(file_path);
+        } else {
+            panic!("Could not find '{}' in {:?}", file, path);
+        }
+    }
+
+    found
+}
+
+fn make_win_dist(rust_root: &Path, plat_root: &Path, target_triple: &str) {
+    //Ask gcc where it keeps its stuff
+    let gcc_out =
+        String::from_utf8(
+            Command::new("gcc.exe")
+                .arg("-print-search-dirs")
+                .output()
+                .expect("failed to execute gcc")
+                .stdout).expect("gcc.exe output was not utf8");
+    let mut bin_path: Vec<_> =
+        env::split_paths(&env::var_os("PATH").expect("failed to get $PATH"))
+        .collect();
+    let mut lib_path = Vec::new();
+
+    for line in gcc_out.lines() {
+        let idx = line.find(':').unwrap();
+        let key = &line[..idx];
+        let trim_chars: &[_] = &[' ', '='];
+        let mut value: Vec<_> =
+            line[(idx + 1)..]
+                .trim_left_matches(trim_chars)
+                .split(';')
+                .map(|s| PathBuf::from(s))
+                .collect();
+
+        if key == "programs" {
+            bin_path.append(&mut value);
```

Comment:
You can probably use `extend` here actually to avoid the `collect` above, dropping a few lines in the process!